b9lab-content/3-main-concepts/05-transactions.md

```diff
@@ -1 +1,162 @@
+
+## Messages
+
+<HighlightBox type=”info”>
+Transaction messages are not to be confused with ABCI messages that define interactions between Tendermint and application layers.
+</HighlightBox>
+
+In this context, messages, or `sdk.Msgs` are module-specific objects that trigger state transitions within the scope of the module they belong to. Module developers define module messages by adding methods to the Protobuf Msg service and defining a `MsgServer`. Each `sdk.Msgs` is related to exactly one Protobuf Msg service RPC, defined inside each module's `tx.proto` file. An sdk app router automatically maps every `sdk.Msg` to a corresponding RPC service which routes it to the appropriate method. Protobuf generates a MsgServer interface for each module Msg service, and the module developer implements this interface.
+
+This design puts more responsibility on module developers, allowing application developers to reuse common functionalities without having to implement state transition logic repetitively.
+While messages contain the information for state transition logic, a transaction's other metadata and relevant information are stored in the `TxBuilder` and `Context`.
+
+## Transaction Generation
+
+The TxBuilder interface contains metadata closely related to the generation of transactions. The end-user can freely set these parameters for the transaction to be generated.
+
+* `Msgs`: the array of messages included in the transaction.
+* `GasLimit`: option chosen by the users for how to calculate how much gas they are willing to spend.
+* `Memo`: a note or comment to send with the transaction.
+* `FeeAmount`: the maximum amount the user is willing to pay in fees.
+* `TimeoutHeight`:, block height until which the transaction is valid.
+* `Signatures`: the array of signatures from all signers of the transaction.
+
+As there are currently two modes for signing transactions, there are also two implementations of TxBuilder. There is a wrapper for `SIGN_MODE_DIRECT` and the `StdTxBuilder` for `SIGN_MODE_LEGACY_AMINO_JSON`. However, the two possibilities should normally be hidden away because end-users should prefer the overarching `TxConfig` interface. `TxConfig` is an app-wide configuration for managing transactions. Most importantly, it holds the information about whether to sign each transaction with SIGN_MODE_DIRECT or SIGN_MODE_LEGACY_AMINO_JSON.
+
+By calling `txBuilder := txConfig.NewTxBuilder()`, a new TxBuilder will be created with the appropriate sign mode.Once TxBuilder is correctly populated with the setters described above, `TxConfig` will also take care of correctly encoding the bytes (again, either using `SIGN_MODE_DIRECT` or `SIGN_MODE_LEGACY_AMINO_JSON`).
+
+Here's a pseudo-code snippet of how to generate and encode a transaction, using the TxEncoder() method:
+
+```
+txBuilder := txConfig.NewTxBuilder()
+txBuilder.SetMsgs(...) // and other setters on txBuilder
+```
+## Broadcast the Transaction
```

Comment:
I see there's no section on how to sign a generated tx using a priv key. I would say that's relevant to this document, but is there a plan to put it somewhere else?